### PR TITLE
Use mkdirp instead of mkdir to handle creating parent directories

### PR DESCRIPTION
**Pull request by davidhao3300**
Because we allow custom data directories, we now face an issue where we attempt to create a directory whose parent directory hasn't been created yet. Use mkdirp to handle creating parent directories.
If fs.mkdir = mkdir, then mkdirp = mkdir -p

### CODE REVIEW

**Comment by johanneswuerbach** [comment]
Thanks, could you add a test? Seems like an easy regression.

**Comment by davidhao3300** [comment]
done

File: tests/config_tests.js [hunk 1]
```diff
@@ -246,14 +247,19 @@ describe('Config', function() {
...
-      var correctFlag = config.cwd() + '/customDirectory/testem.firefox';
+      var correctFlag = config.cwd() + '/node_modules/testem.firefox';
       if (process.platform === 'win32') {
-        correctFlag = config.cwd() + '\\customDirectory\\testem.firefox';
+        correctFlag = config.cwd() + '\\node_modules\\testem.firefox';
       }
       expect(launchers.firefox.settings.args).to.include(correctFlag);
-      done();
+      launchers.firefox.settings.setup(config, function() {
+        fs.exists(correctFlag, function(exists) {
```
Comment (johanneswuerbach) [inline tests/config_tests.js]: Hm, `fs.exists` is deprecated https://nodejs.org/api/fs.html#fs_fs_exists_path_callback, could you use https://nodejs.org/api/fs.html#fs_fs_stat_path_callback instead?

**Comment by johanneswuerbach** [comment]
small nit, otherwise :+1:

**Comment by davidhao3300** [comment]
done

**Pull request merged**